internal/providers/terraform/aws/eks_node_group.go

```diff
@@ -70,14 +70,28 @@ func NewEKSNodeGroup(d *schema.ResourceData, u *schema.UsageData) *schema.Resour
 	}
 }
 
-func eksComputeCostComponent(d *schema.ResourceData, region string, desiredSize int64, instanceType string) *schema.CostComponent {
-
+func eksComputeCostComponent(d *schema.ResourceData, u *schema.UsageData, region string, desiredSize int64, instanceType string) *schema.CostComponent {
 	purchaseOptionLabel := "on_demand"
-
 	if d.Get("capacity_type").String() != "" {
 		purchaseOptionLabel = strings.ToLower(d.Get("capacity_type").String())
 	}
 
+	var RIType, RITerm, RIPaymentOption string
+	if u != nil && u.Get("reserved_instance_type").Exists() {
+		purchaseOptionLabel = "reserved"
+		RIType = u.Get("reserved_instance_type").String()
+		if u.Get("reserved_instance_term").Exists() {
+			RITerm = u.Get("reserved_instance_term").String()
+		}
+		if u.Get("reserved_instance_payment_option").Exists() {
+			RIPaymentOption = u.Get("reserved_instance_payment_option").String()
+		}
+	}
+
+	if RIType != "" {
```

Comment:
Should this also be checking that RITerm and RIPaymentOption are set? otherwise the price lookup might return multiple prices if those are not set?

Reply:
Yes, sure. Will add checks for them too